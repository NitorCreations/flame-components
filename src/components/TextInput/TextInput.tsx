import * as React from 'react';

interface IInputProps {
  placeholder?: string;
  label?: string;
  onChange?: () => void;
  type?: string;
  disabled?: boolean;
  className?: string;
}

export default class TextInput extends React.PureComponent<IInputProps, {}> {
  static defaultProps = {
    type: 'text'
  };

  public render() {
    const { onChange, placeholder, label, type, className, disabled } = this.props;

    const labelBlock = label ? (
      <span className="flame-input-label">
          {label}
        </span>
    ) : null;

    return (
      <div className={`flame-input ${className ? className : ''}`}>
        {labelBlock}
        <input
          type={type}
          placeholder={placeholder}
          disabled={disabled}
          onChange={onChange}
        />
      </div>
    );
  }
}