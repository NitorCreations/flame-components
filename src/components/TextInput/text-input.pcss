.flame-input {
  height: calc(var(--global-baseline) * 4);
  padding-left: var(--global-baseline);
  padding-right: var(--global-baseline);
  border-radius: var(--border-radius);
  border: 2px solid var(--light-gray);
  will-change: box-shadow;
  transition: box-shadow var(--hover-transition);
  &:hover {
    box-shadow: 0 0 4px 0 var(--light-gray);
  }
  &:focus {
    outline: none;
    border: 2px solid var(--nitor-blue-light);
    box-shadow: 0 0 4px 0 var(--nitor-blue-light);
  }
  &:disabled {
    background-color: var(--light-gray);
    border: 2px solid var(--dark-gray);
    cursor: not-allowed;
  }
}