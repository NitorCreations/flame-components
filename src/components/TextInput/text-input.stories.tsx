import * as React from 'react';
import { storiesOf } from '@storybook/react';
import TextInput from './TextInput';

storiesOf('TextInput', module)
  .add('Normal', () => (
      <TextInput placeholder="Text input" />
    )
  )
  .add('With label', () => (
      <TextInput placeholder="With label" label="Label" />
    )
  )
  .add('Disabled', () => (
      <TextInput placeholder="Disabled input" disabled />
    )
  );