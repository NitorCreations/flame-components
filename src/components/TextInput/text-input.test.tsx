import * as React from 'react';
import { shallow } from 'enzyme';
import TextInput from './TextInput';

describe('TextInput', () => {
  it('Should render correctly', () => {
    const component = shallow(<TextInput />);

    expect(component).toMatchSnapshot();
  });

  it('Should be correctly enabled', () => {
    const component = shallow(<TextInput disabled={false} />);

    expect(component).toMatchSnapshot();
  });

  it('Should be correctly disabled', () => {
    const component = shallow(<TextInput disabled />);

    expect(component).toMatchSnapshot();
  });

  it('Should render label', () => {
    const component = shallow(<TextInput label="label" />);

    expect(component).toMatchSnapshot();
  });
});