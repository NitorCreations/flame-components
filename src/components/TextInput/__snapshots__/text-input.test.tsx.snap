// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Should be correctly disabled 1`] = `
<div
  className="flame-input "
>
  <input
    disabled={true}
    type="text"
  />
</div>
`;

exports[`TextInput Should be correctly enabled 1`] = `
<div
  className="flame-input "
>
  <input
    disabled={false}
    type="text"
  />
</div>
`;

exports[`TextInput Should render correctly 1`] = `
<div
  className="flame-input "
>
  <input
    type="text"
  />
</div>
`;

exports[`TextInput Should render label 1`] = `
<div
  className="flame-input "
>
  <span
    className="flame-input-label"
  >
    label
  </span>
  <input
    type="text"
  />
</div>
`;
